testsuite/integration-arquillian/tests/base/src/test/java/org/keycloak/testsuite/forms/LevelOfAssuranceFlowTest.java

```diff
@@ -0,0 +1,227 @@
+package org.keycloak.testsuite.forms;
+
+import static org.keycloak.testsuite.admin.AbstractAdminTest.loadJson;
+
+import java.net.URLEncoder;
+import java.nio.charset.StandardCharsets;
+import java.util.Collections;
+import java.util.Iterator;
+import java.util.List;
+
+import javax.ws.rs.core.UriBuilder;
+
+import org.jboss.arquillian.graphene.page.Page;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.keycloak.authentication.authenticators.browser.PasswordFormFactory;
+import org.keycloak.authentication.authenticators.browser.SetLoaAuthenticatorFactory;
+import org.keycloak.authentication.authenticators.browser.UsernameFormFactory;
+import org.keycloak.authentication.authenticators.conditional.ConditionalLoaAuthenticatorFactory;
+import org.keycloak.common.Profile;
+import org.keycloak.events.Details;
+import org.keycloak.models.AuthenticationExecutionModel.Requirement;
+import org.keycloak.representations.idm.RealmRepresentation;
+import org.keycloak.testsuite.AbstractTestRealmKeycloakTest;
+import org.keycloak.testsuite.AssertEvents;
+import org.keycloak.testsuite.arquillian.annotation.EnableFeature;
+import org.keycloak.testsuite.authentication.PushButtonAuthenticatorFactory;
+import org.keycloak.testsuite.pages.ErrorPage;
+import org.keycloak.testsuite.pages.LoginUsernameOnlyPage;
+import org.keycloak.testsuite.pages.PasswordPage;
+import org.keycloak.testsuite.util.FlowUtil;
+import org.openqa.selenium.By;
+
+/**
+ * Tests for Level Of Assurance conditions in authentication flow.
+ * 
+ * @author <a href="mailto:sebastian.zoescher@prime-sign.com">Sebastian Zoescher</a>
+ */
+@EnableFeature(value = Profile.Feature.WEB_AUTHN, skipRestart = true, onlyForProduct = true)
+public class LevelOfAssuranceFlowTest extends AbstractTestRealmKeycloakTest {
```

Comment:
This test is very useful. But I think we should have some more such as:

* Check LoA mapping and how `acr` claim is set in the ID Token
* Check the behavior when the user is authenticated and the client requires a different LoA (step-up).

Also, why we are using `EnableFeature` here and using webauthn feature ?

Reply:
@pedroigor We wanted to provide a few integration tests for the PR, but these can definitely be extended further. Our intention was to provide more integration tests after the code review, when we have a final draft of the code.

And we will remove the EnableFeature annotation as it is not needed.

Reply:
I removed the EnableFeature annotation.